### ISSUE TITLE
Error message when running `bundle env`

### ISSUE THREAD

**Issue by deivid-rodriguez** [bundler/bundler#6152]
When I run `bundle env`, I get the following error before getting the actual env information. No matter whether I'm on a git repo or not.

```
fatal: Not a git repository (or any parent up to mount point /)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
```

## Environment

```
Bundler       1.16.0
  Platforms   ruby, x86_64-linux
Ruby          2.4.2p198 (2017-09-14 revision 59899) [x86_64-linux]
  Full Path   /home/deivid/.rubies/ruby-2.4.2/bin/ruby
  Config Dir  /home/deivid/.rubies/ruby-2.4.2/etc
RubyGems      2.7.1
  Gem Home    /home/deivid/.gem/ruby/2.4.2
  Gem Path    /home/deivid/.gem/ruby/2.4.2:/home/deivid/.rubies/ruby-2.4.2/lib/ruby/gems/2.4.0
  User Path   /home/deivid/.gem/ruby/2.4.0
  Bin Dir     /home/deivid/.gem/ruby/2.4.2/bin
Tools         
  Git         2.15.0
  RVM         not installed
  rbenv       not installed
  chruby      0.3.9
```

## Bundler Build Metadata

```
Built At          2017-11-05
Git SHA           
Released Version  false
```

## Bundler settings

```
gem.test
  Set for the current user (/home/deivid/.bundle/config): "rspec"
gem.mit
  Set for the current user (/home/deivid/.bundle/config): true
gem.coc
  Set for the current user (/home/deivid/.bundle/config): true
```

**Comment by deivid-rodriguez** [comment]
The error is the same as in https://github.com/rubygems/rubygems/issues/2059

**Comment by maximeg** [comment]
Same thing with `bundle version`

```
➜  server git:(master) ✗  2.4.2 bundle version
fatal: Not a git repository (or any of the parent directories): .git
Bundler version 1.16.0 (2017-11-19 commit )
```

Looks like the error is in `Bundler::BuildMetadata.git_commit_sha`.
Somehow, having bundler coming with rubygems make things different.

**Comment by colby-swandale** [comment]
This looks like this issue has been fixed with RubyGems 2.7.2
